Sources/Client/Client/Client+Config.swift

```diff
@@ -35,18 +37,22 @@ extension Client {
         ///     - database: a database manager.
         ///     - callbackQueue: a request callback queue, default nil (some background thread).
         ///     - logOptions: enable logs (see `ClientLogger.Options`), e.g. `.all`
+        ///     - defaultURLSessionConfiguration: the URLSessionConfiguration the Client uses as
+        ///     the default configuration.
         public init(apiKey: String,
                     baseURL: BaseURL = .usEast,
                     stayConnectedInBackground: Bool = true,
                     database: Database? = nil,
                     callbackQueue: DispatchQueue? = nil,
-                    logOptions: ClientLogger.Options = []) {
+                    logOptions: ClientLogger.Options = [],
+                    defaultURLSessionConfiguration: URLSessionConfiguration = .default) {
```

Comment:
I'm not sure if this should be configurable by the user. We're not going to be able to make some assumptions about session configuration when we make this configurable. If there's no use case for users, I'd say KISS

Reply:
Good point, I didn't realize this is a user-facing config.

Reply:
I agree with @b-onc It should be internal configuration.

Reply:
I'll fix this once we merge https://github.com/GetStream/stream-chat-swift/pull/231 since it touches the same part of the codebase.

Reply:
fixed